src/OrchardCore.Modules/OrchardCore.Title/Handlers/TitlePartHandler.cs

```diff
@@ -56,6 +57,14 @@ private async Task SetTitleAsync(TitlePart part)
         {
             var settings = GetSettings(part);
 
+            if (_contentItemIds.Contains(part.ContentItem.ContentItemId))
+            {
+                return;
+            }
+
+            // Save the processed contentItemId to prevent it from being processed more than once.
+            _contentItemIds.Add(part.ContentItem.ContentItemId);
```

Comment:
You can just use `if (!_contentItemIds.Add()`

Also you can use an `HashSet<ContentItem> _contentItems`

Reply:
Was just for info, as we already do in some places.
It will compare by reference, so a few bytes to compare.

Reply:
Good to know that reference check is cheaper that string comparison. Made the change. Let me know if you are good with this PR so we can merge it before 1.6.

Reply:
Yes, but can't be used everywhere, needs to be sure that we are dealing with the same instances.

Or that the type is for example an `IEquatable` that implement a cheap comparison.